lib/Makefile

```diff
@@ -238,7 +246,11 @@ ZSTD_STATLIB_OBJ := $(addprefix $(ZSTD_STATLIB_DIR)/,$(ZSTD_LOCAL_OBJ))
 $(ZSTD_STATLIB): ARFLAGS = rcs
 $(ZSTD_STATLIB): | $(ZSTD_STATLIB_DIR)
 $(ZSTD_STATLIB): $(ZSTD_STATLIB_OBJ)
-	@echo compiling static library
+ifneq (,$(filter -DZSTD_MULTITHREAD,$(CPPFLAGS)))
+	@echo compiling multi-threaded static library $(LIBVER)
+else
+	@echo compiling single-threaded static library $(LIBVER)
+endif
```

Comment:
Bizzare behavior I don't understand here: When I compile with `BUILD_DIR` set, somehow the expression `$(filter -DZSTD_MULTITHREAD,$(CPPFLAGS))` evaluates to empty string. This causes us to output `compiling single-threaded` even in mt builds.

The confusing thing is that even when I also `@echo $(CPPFLAGS)` in the same line, it clearly shows that `-DZSTD_MULTITHREAD` is in `CPPFLAGS`, and even `@echo $(filter -DZSTD_MULTITHREAD,$(CPPFLAGS))` evaluates to `-DZSTD_MULTITHREAD`. Is there something weird going on with the `SET_CACHE_DIRECTORY` thing?

Reply:
Yep, it was a question of parsing order,
with the initial formulation being part of stage 1.
Since `BUILD_DIR` skips reloading the `Makefile`,
all updates, which are part of stage 2, are ignored for stage 1 
(without `BUILD_DIR`, `Makefile` is reloaded, and variables are passed to following process,
hence "stage 2" changes become "stage 1" for the following process).

The new formulation is part of stage 3,
thus applies any change achieved during previous stages.